### PR TITLE
Cleanup of calendar model, show calendar by default if owned by user

### LINKED ISSUE #1588: The new created calendars are not activated by default
Do we need to set `calendar-enabled` on calendar app side or is this an issue on server side?

### PR DESCRIPTION
**Pull request by georgehrke**
fixed #1588 

Cleans up calendar model and adds tests

### CODE REVIEW

File: src/models/calendar.js [hunk 1]
```diff
@@ -75,39 +74,92 @@ export const getDefaultCalendarObject = (props = {}) => Object.assign({}, {
...
-	if (color.length === 9) {
-		// Make sure it's #RRGGBB, not #RRGGBBAA
-		color = color.substr(0, 7)
+export function mapDavCollectionToCalendar(calendar, currentUserPrincipal) {
+	const id = btoa(calendar.url)
+	const displayName = calendar.displayname
+
+	// calendar.color can be set to anything on the server,
+	// so make sure it's something that remotely looks like a color
+	let color
```
Comment (tcitworld) [inline src/models/calendar.js]: This whole part might as well be moved to `utils/color.js` in a `detectColor` function.

File: src/models/calendar.js [hunk 2]
```diff
@@ -119,20 +171,31 @@ export function mapDavCollectionToCalendar(calendar) {
...
+	let displayName
+	if (sharee['common-name']) {
+		displayName = sharee['common-name']
+	} else {
+		if (sharee.href.startsWith('principal:principals/groups/')) {
+			displayName = sharee.href.substr(28)
+		} else if (sharee.href.startsWith('principal:principals/users/')) {
+			displayName = sharee.href.substr(27)
+		} else {
+			displayName = sharee.href
```
Comment (tcitworld) [inline src/models/calendar.js]: No circles handing?

Comment (georgehrke) [reply src/models/calendar.js]: Circles always come with a common-name.

File: src/models/calendar.js (hunk 1 above)
Comment (georgehrke) [reply src/models/calendar.js]: Done 👍

**Pull request merged**